tests/test_pattern.py

```diff
@@ -33,6 +37,17 @@ def compare_backend_result_with_statevec(backend: str, backend_state, statevec:
         raise NotImplementedError(backend)
 
 
+Outcome = typing.Literal[0, 1]
+
+
+class IterGenerator:
+    def __init__(self, it: typing.Iterator[Outcome]) -> None:
+        self.__it = it
```

Comment:
`Iterator` cannot be reused: this means that `IterGenerator` takes the ownership of `it` provided by caller.
Is it really what you wanted?

Reply:
I acknowledge it is safer to take an `Iterable` and use `iter` on it. Fixed in 41a9123.